pkg/internal/computeresources/tasklevel/tasklevel_test.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tasklevel_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/internal/computeresources/tasklevel"
+	"github.com/tektoncd/pipeline/test/diff"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func TestPodBuild_TaskLevelResourceRequirements(t *testing.T) {
+	testcases := []struct {
+		desc                     string
+		ts                       v1beta1.TaskSpec
+		trs                      v1beta1.TaskRunSpec
+		expectedComputeResources []corev1.ResourceRequirements
+	}{{
+		desc: "only with requests",
+		ts: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "1st-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}, {
+				Name:    "2nd-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}},
+		},
+		trs: v1beta1.TaskRunSpec{
+			ComputeResources: &corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2")},
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+		}, {
+			Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+		}},
+	}, {
+		desc: "only with limits",
+		ts: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "1st-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}, {
+				Name:    "2nd-step",
+				Image:   "image",
+				Command: []string{"cmd"},
+			}},
+		},
+		trs: v1beta1.TaskRunSpec{
+			ComputeResources: &corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+			},
+		},
+		expectedComputeResources: []corev1.ResourceRequirements{{
+			Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
+		}, {
+			Limits: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("500m")},
```

Comment:
thanks for more thought on this point. 

I tend to take the "request as the existing limit when only limit is configured" way
- think will align with current k8s behaviours - so ease user's understanding and usages
- and "0 / smallest as possible" request could lead to a "resource hungry" case in the runtime, like the CPU requests acting as weights to acquire CPU time as https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#how-pods-with-resource-limits-are-run

Reply:
Good point that this could lead to a pod being more likely to be evicted. I'll update the TEP. in the meantime, I think you can go ahead with implementing this strategy.

Reply:
https://github.com/tektoncd/community/pull/776

Reply:
Unresolving since I don't think this comment was addressed; just to be clear I'm suggesting implementing what's proposed in the TEP PR.

Reply:
thought this will be handled in limitrange (wrongly). but a case will be if there is no limitrange.
let me update

Reply:
added the required logic and updated testcases.